src/main/java/org/araymond/joal/core/client/emulated/generator/key/TimedRefreshKeyGenerator.java

```diff
@@ -49,7 +52,11 @@ public String getKey(final InfoHash infoHash, final RequestEvent event) {
         return this.key;
     }
 
-    private boolean shouldRegenerateKey() {
+    boolean shouldRegenerateKey() {
         return this.lastGeneration == null || ChronoUnit.SECONDS.between(this.lastGeneration, LocalDateTime.now()) >= this.refreshEvery;
     }
+
+    void setKey(String key) {
+        this.key = key;
+    }
```

Comment:
Not entirely happy about this, but allows `TimedOrAfterStartedAnnounceRefreshKeyGenerator` to inherit this class.

Reply:
Agreed, but it makes sense for inheritence